manta-accounting/src/wallet/signer/functions.rs

```diff
@@ -774,6 +808,137 @@ where
     Ok(into_array_unchecked(final_presenders))
 }
 
+/// Performs a ToPublic transaction spending the assets in `selection`,
+/// returning [`TransferPost`]s.
+#[allow(clippy::too_many_arguments)]
+#[inline]
+fn compute_to_public_transaction<C>(
+    accounts: &AccountTable<C>,
+    assets: &C::AssetMap,
+    utxo_accumulator: &mut C::UtxoAccumulator,
+    parameters: &Parameters<C>,
+    proving_context: &MultiProvingContext<C>,
+    asset_id: &C::AssetId,
+    sink_accounts: Vec<C::AccountId>,
+    selection: Selection<C>,
+    rng: &mut C::Rng,
+) -> Result<SignResponse<C>, SignError<C>>
+where
+    C: Configuration,
+    C::AssetValue: SubAssign,
+{
+    let Selection {
+        mut change,
+        mut pre_senders,
+    } = selection;
+    let mut posts = Vec::new();
+    let mut iter = pre_senders
+        .into_iter()
+        .chunk_by::<{ ToPublicShape::SENDERS }>();
+    for chunk in &mut iter {
+        let senders = array_map(chunk, |s| {
+            s.try_upgrade(parameters, utxo_accumulator)
+                .expect("Unable to upgrade expected UTXO.")
+        });
+        process_to_public_senders(
+            accounts,
+            utxo_accumulator,
+            parameters,
+            proving_context,
+            asset_id,
+            &mut change,
+            senders,
+            sink_accounts.clone(),
+            &mut posts,
+            rng,
+        )?;
+    }
+    pre_senders = iter.remainder();
+    if !pre_senders.is_empty() {
+        let final_senders = into_array_unchecked(prepare_final_pre_senders(
+            accounts,
+            assets,
+            utxo_accumulator,
+            parameters,
+            asset_id,
+            Default::default(),
+            pre_senders,
+            rng,
+        )?);
+        process_to_public_senders(
+            accounts,
+            utxo_accumulator,
+            parameters,
+            proving_context,
+            asset_id,
+            &mut change,
+            final_senders,
+            sink_accounts,
+            &mut posts,
+            rng,
+        )?;
+    }
+    Ok(SignResponse::new(posts))
+}
+
+/// Creates a to public [`TransferPost`] spending the assets held by `senders` and
+/// attaches it to `post`.
+#[allow(clippy::too_many_arguments)]
+#[inline]
+fn process_to_public_senders<C>(
+    accounts: &AccountTable<C>,
+    utxo_accumulator: &mut C::UtxoAccumulator,
+    parameters: &Parameters<C>,
+    proving_context: &MultiProvingContext<C>,
+    asset_id: &C::AssetId,
+    change: &mut C::AssetValue,
+    senders: [Sender<C>; ToPublicShape::SENDERS],
+    sink_accounts: Vec<C::AccountId>,
+    posts: &mut Vec<TransferPost<C>>,
+    rng: &mut C::Rng,
+) -> Result<(), SignError<C>>
+where
+    C: Configuration,
+    C::AssetValue: SubAssign,
+{
+    let authorization = authorization_for_default_spending_key::<C>(accounts, parameters, rng);
+    let mut received_value = C::AssetValue::default();
```

Comment:
What's the alternative? Remember at the `manta-accounting` abstraction level it doesn't make sense to write `received_value = 0` (because we don't know yet it will be a `u128` or any other integer type for that matter).